packages/create-astro/src/actions/context.ts

```diff
@@ -83,9 +80,24 @@ export async function getContext(argv: string[]): Promise<Context> {
 	}
 
 	skipHouston =
-		((os.platform() === 'win32' && !fancy) || skipHouston) ??
+		((os.platform() === 'win32' && !fancy) || !!skipHouston) ??
 		[yes, no, install, git, typescript].some((v) => v !== undefined);
 
+	// We use `strict: false` in `parseArgs` to allow unknown options, but Node also
+	// simply doesn't guarantee the types anymore, so we need to validate ourselves :(
+	help = !!help;
+	template = typeof template == 'string' ? template : undefined;
+	no = !!no;
+	yes = !!yes;
+	install = !!install;
+	noInstall = !!noInstall;
+	git = !!git;
+	noGit = !!noGit;
+	typescript = typeof typescript == 'string' ? typescript : undefined;
+	fancy = !!fancy;
+	dryRun = !!dryRun;
+	ref = typeof ref == 'string' ? ref : undefined;
```

Comment:
One of the pain points with `parseArgs` is that if we want to allow unknown flags or positionals to be passed, it completely skips flag validation as well, so the `type` we specify for each flag is largely useless.

Anyways maybe we can consider being strict and not allow unknown options in the future, but that'll be a breaking change

Reply:
Would something like `typescript ||= undefined` work here? Also I don't know what style guide we have for it, but I think an explicit `Boolean()` call is clearer than `!!`

Reply:
> Anyways maybe we can consider being strict and not allow unknown options in the future, but that'll be a breaking change

IMHO, we should send this PR to `next` and break the behaviour of the CLI to report an error for unknown flags.

Reply:
> Would something like `typescript ||= undefined` work here?

The initial types for `typescript` is `string | boolean | undefined`. The line narrows it to `string | undefined`. So I'm not sure a `||=` would help here if the value is eg `true`.

> I think an explicit `Boolean()` call is clearer than `!!`

I'm team `!!` as it's shorter and [slightly faster](https://jsperf.app/woguyu) 😄 

> IMHO, we should send this PR to `next` and break the behaviour of the CLI to report an error for unknown flags.

Couldn't we do the `parseArgs` migration on main first, then do the breaking change in `next`?

Reply:
I'll take out the trash as soon as this change goes into next 😄